File: example/python/eva_once.py
import sys, time, uuid, base64
import requests
from Signer import Signer 

class EvaClient:
    def __init__(self, appId, appSecret, audioBuffer, sText, session):
        self.appId = appId
        self.appSecret = appSecret
        self.audioBuffer = audioBuffer
        self.schema = "https"
        self.method = "post"
        self.host = "smartedu.html5.qq.com"
        self.path = "/v1/paragraph_follow_once"
        self.sText = sText
        self.session = session

    def send(self):
        uuidv1 = uuid.uuid1()
        print(uuidv1)
        timestamp = int(round(time.time() * 1000))
        bodyParams = {
                "sAudio": base64.b64encode(self.audioBuffer).decode(),
                "sSessionId": uuidv1,
                "iAudioFormat": 146,#0x5000000,
                "sText":self.sText,
                "sGuid":"test-guid"
            }
        signer = Signer(self.appId, self.appSecret, timestamp, self.method, self.host, self.path, {}, bodyParams )
        authHeader = signer.getAuthorizationHeader()
        response = self.session.post(url="{}://{}{}".format(self.schema, self.host, self.path),data=bodyParams,headers={"Authorization": authHeader})
        now = int(round(time.time() * 1000))
        print(now - timestamp , response.elapsed.microseconds, response.text)

if __name__ == "__main__":
    # 注意：使用此方法以开启http的keep-alive选项，能避免多次建立、断开tcp连接的开销，大幅提升接口性能
    # http的keep-alive，各语言实现方式不同，但都应该开启
    session = requests.session()
    with open("./test.wav", "rb") as audioBuffer:
         asrClient = EvaClient("you appid", "your appsecret", audioBuffer.read(),"how are you", session)
         asrClient.send()

File: example/nodejs/eva_once.js
const fsPromise = require("fs").promises
const uuidv1 = require("uuid").v1
const EduSigner = require("./Signer")
const request = require("request")

class EvaClient{
    constructor(appId, appSecret,audioBuffer, sText){
        this.appId = appId
        this.appSecret = appSecret
        this.sText = sText
        this.audioBuffer = audioBuffer
        this.schema = "https"
        this.method = "post"
        this.host = "smartedu.html5.qq.com"
        this.path = "/v1/sentence_follow_once"
    }

    async sendOnce(){
        return new Promise((resolve, reject)=>{
            let body = {
                sGuid: "czzou_test",
                sText: this.sText,
                iAudioFormat: 0x92,
                sAudio: this.audioBuffer.toString("base64"),
                sSessionId: uuidv1()
            }
            let timestamp = new Date().getTime()
            
            let signer = new EduSigner(this.appId,this.appSecret,timestamp, this.method,this.host,this.path,{},body)
            request.post(`${this.schema}://${this.host}${this.path}`, {
                form:body,
                headers:{
                    Authorization:`algorithm=sha256&appid=${this.appId}&timestamp=${timestamp}&sig=${signer.signature}`
                },
                json: true
            }, (err, response)=>{
                if(err){
                    reject(err)
                } else {
                    resolve(response.body)
                }
            })
        })
    }

}

async function test(){
    try{
        const audioBuffer = await fsPromise.readFile("test.wav")
        asrClient = new EvaClient("your appid", "your appkey", audioBuffer,"how are you")
        let response = await asrClient.sendOnce()
        console.log(response)
    } catch(e){
        console.log("send error:", e)
    }
}

test()